src/nbl/ui/CWindowWin32.cpp

```diff
@@ -369,6 +367,120 @@ namespace ui
+					lk.lock();
+					inputChannel->pushIntoBackground(std::move(event));
+				}
+				if (rawMouse.usButtonFlags & RI_MOUSE_LEFT_BUTTON_DOWN)
+				{
+					SMouseEvent event;
+					event.type = SMouseEvent::EMT_BUTTON;
+					event.buttonEvent.state = SMouseEvent::SButtonEvent::ES_PRESSED;
+					event.buttonEvent.button = ui::E_MOUSEBUTTON::EMB_LEFT_BUTTON;
+					auto lk = inputChannel->lockBackgroundBuffer();
+					lk.lock();
+					inputChannel->pushIntoBackground(std::move(event));
+				}
+				else if (rawMouse.usButtonFlags & RI_MOUSE_LEFT_BUTTON_UP)
+				{
+					SMouseEvent event;
+					event.type = SMouseEvent::EMT_BUTTON;
+					event.buttonEvent.state = SMouseEvent::SButtonEvent::ES_RELEASED;
+					event.buttonEvent.button = ui::E_MOUSEBUTTON::EMB_LEFT_BUTTON;
+					auto lk = inputChannel->lockBackgroundBuffer();
+					lk.lock();
+					inputChannel->pushIntoBackground(std::move(event));
+				}
+				if (rawMouse.usButtonFlags & RI_MOUSE_RIGHT_BUTTON_DOWN)
+				{
+					SMouseEvent event;
+					event.type = SMouseEvent::EMT_BUTTON;
+					event.buttonEvent.state = SMouseEvent::SButtonEvent::ES_PRESSED;
+					event.buttonEvent.button = ui::E_MOUSEBUTTON::EMB_RIGHT_BUTTON;
+					auto lk = inputChannel->lockBackgroundBuffer();
+					lk.lock();
+					inputChannel->pushIntoBackground(std::move(event));
+				}
+				else if (rawMouse.usButtonFlags & RI_MOUSE_RIGHT_BUTTON_UP)
+				{
+					SMouseEvent event;
+					event.type = SMouseEvent::EMT_BUTTON;
+					event.buttonEvent.state = SMouseEvent::SButtonEvent::ES_RELEASED;
+					event.buttonEvent.button = ui::E_MOUSEBUTTON::EMB_RIGHT_BUTTON;
+					auto lk = inputChannel->lockBackgroundBuffer();
+					lk.lock();
+					inputChannel->pushIntoBackground(std::move(event));
+				}
+				if (rawMouse.usButtonFlags & RI_MOUSE_MIDDLE_BUTTON_DOWN)
+				{
+					SMouseEvent event;
+					event.type = SMouseEvent::EMT_BUTTON;
+					event.buttonEvent.state = SMouseEvent::SButtonEvent::ES_PRESSED;
+					event.buttonEvent.button = ui::E_MOUSEBUTTON::EMB_MIDDLE_BUTTON;
+					auto lk = inputChannel->lockBackgroundBuffer();
+					lk.lock();
+					inputChannel->pushIntoBackground(std::move(event));
+				}
+				else if (rawMouse.usButtonFlags & RI_MOUSE_MIDDLE_BUTTON_UP)
+				{
+					SMouseEvent event;
+					event.type = SMouseEvent::EMT_BUTTON;
+					event.buttonEvent.state = SMouseEvent::SButtonEvent::ES_RELEASED;
+					event.buttonEvent.button = ui::E_MOUSEBUTTON::EMB_MIDDLE_BUTTON;
+					auto lk = inputChannel->lockBackgroundBuffer();
+					lk.lock();
+					inputChannel->pushIntoBackground(std::move(event));
+				}
+				// TODO other mouse buttons
+
+				if (rawMouse.usButtonFlags & RI_MOUSE_WHEEL)
+				{
+					SHORT wheelDelta = static_cast<SHORT>(rawMouse.usButtonData);
+					SMouseEvent event;
+					event.type = SMouseEvent::EMT_WHEEL;
+					event.motionEvent.verticalDelta = wheelDelta;
+					event.motionEvent.horizontalDelta = 0; // TODO horizontal wheel (lol never seen one)
```

Comment:
vertical move of two fingers on laptop's touchpad for example

Reply:
Ah yeah, forgot that one